src/theory/arith/theory_arith.cpp

```diff
@@ -292,8 +293,7 @@ bool TheoryArith::collectModelValues(TheoryModel* m,
     {
       continue;
     }
-    // maps to constant of comparable type
-    Assert(p.first.getType().isComparableTo(p.second.getType()));
+    // maps to constant of same type
```

Comment:
Why not keep the assertion, or make it stricter that the types should be identical?

Reply:
Whoops, I have the strict version on my branch, and removed it from this PR, I didnt realize the weaker version is already there.

I cannot make it strict until the next PR.